### DISCUSSION TITLE
Support displaying uploaded videos for `video` field

### DISCUSSION THREAD

**Discussion by susanu** [Laravel-Backpack/community-forum#711] [Ideas]
Hi guys,

Would be a great idea if we would get a similar field to `image` for `video`.
I have some fields that require video uploads and it would be great if i can see the video uploaded.

Thank you!

**Comment by pxpm** [comment]
Hey @susanu thanks for the suggestion.

What kind of video upload are we talking about?  

An youtube/vimeo/etc upload link (like in our video field/column), or actually uploading `.videos` to the server ?

If the later, it would need to be a field/column that embed some kind of JS video player right ? 

And for it to work with a "preview on upload", it needs to be and ajax uploader or some JS streamer ?

Cheers

**Comment by susanu** [comment]
I usually upload mp4 files into an s3 bucket and i want to display the uploaded video next to the field or under (like the `image` field)
If the disk supports `url` or `temporaryUrl` a new `video` html element would be displayed under the field.

No js needed, just a `video` element.

**Reply by pxpm** [reply]
That can be achieved by copying our `upload` field into `video_upload` and adding a preview under the field right ? 

```html

// upload field code 

@if(isset($field['value'])
<video ... />
@endif
```

I am not sure we should create a `video_upload` now, and then comes the `pdf_upload` and the `xxx_upload`. 

Maybe ... just maybe if we generalize it like `upload_with_preview` that has some built in previews for the most common types (img, video .. ??? ) and allow you to create custom previews for types that we don't support ... maybe it makes this feature more "appealing". 

What do you think ? 

Cheers

**Reply by susanu** [reply]
Hi @pxpm,

> Maybe ... just maybe if we generalize it like upload_with_preview that has some built in previews for the most common types (img, video .. ??? ) and allow you to create custom previews for types that we don't support ... maybe it makes this feature more "appealing".

I love the idea but i think this should be an option called ```showPreview``` so we don't end up creating 2 new fields ```upload_with_preview``` and ```upload_multiple_with_preview```.
Also, this option should be available for the columns as well.

What do you think?

**Reply by pxpm** [reply]
Maybe we can modify the `url` column to display accordingly to url type ? 
If url ends with `.mp3` add an `<audio>`, `.mp4` a `<video>`, images to `<img>` ? 

Would that be a "contained" solution that we can push as a non-breaking change and with change only in one file ?

**Reply by susanu** [reply]
I don't see how this would solve my problem.
You don't like the idea with ```showPreview``` on the upload field/column?

**Reply by pxpm** [reply]
I would probably be ok with a showPreview in `column`, but I have many concerns adding it to the field too. 

We can do baby steps, because most likely the code needed for the column preview would fit the field preview anyway.

**Comment by susanu** [comment]
The preview in field is crucial for me because i work a LOT with media (videos, images, audios, documents) in my admin panels.
